crates/subspace-farmer/src/bin/subspace-farmer/commands/cluster/farmer.rs

```diff
@@ -250,14 +262,16 @@ where
                     Arc::clone(&faster_read_sector_record_chunks_mode_concurrency);
 
                 async move {
-                    let farm_fut = SingleDiskFarm::new::<_, _, PosTable>(
+                    let farm_fut = SingleDiskFarm::new::<_, PosTableLegacy, PosTable>(
                         SingleDiskFarmOptions {
                             directory: disk_farm.directory.clone(),
                             farmer_app_info,
                             allocated_space: disk_farm.allocated_space,
                             max_pieces_in_sector,
                             node_client,
                             reward_address,
+                            plotter_legacy,
```

Comment:
Did you consider a wrapper instead of several variables? Imagine we'll have another version.